tests/acceptance/bootstrap/SettingsContext.php

```diff
@@ -596,11 +597,108 @@ public function getBodyForNotificationSetting(string $user): array {
 						"resource" => [
 							"type" => $settingsValue->resource->type
 						],
-						"boolValue" => true
 					]
 				];
 			}
 		}
-		throw new Exception(('`disable-email-notifications` not found in the setting list'));
+		throw new Exception("'$event' not found in the setting list");
+	}
+
+	/**
+	 * @When /^user "([^"]*)" (disables|enables) notification for following event using Settings API:$/
+	 *
+	 * @param string $user
+	 * @param string $enableOrDisable
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function userDisablesNotificationForFollowingEventUsingSettingsApi(
+		string $user,
+		string $enableOrDisable,
+		TableNode $table
+	): void {
+		$settings = $table->getRowsHash();
+		foreach ($settings as $event => $value) {
+			$body = $this->getBodyForNotificationSetting($user, $event);
+			if (str_contains($value, "mail")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "mail","boolValue" => $enableOrDisable === "enables"];
+			}
+			if (str_contains($value, "in-app")) {
+				$body["value"]["collectionValue"]["values"][]
+					= ["key" => "in-app","boolValue" => $enableOrDisable === "enables"];
+			}
+			$response = HttpRequestHelper::sendRequest(
```

Comment:
please, use the existing method
https://github.com/owncloud/ocis/blob/808ea02073022861555bcbda7b73537023fc3351/tests/acceptance/TestHelpers/SettingsHelper.php#L319